modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceIntrospectionTestCase.java

```diff
@@ -50,17 +53,33 @@
     private String consumerSecret;
 
     private CloseableHttpClient client;
+    private final String username;
+    private final String userPassword;
+    private final String activeTenant;
+    private final String TENANT_DOMAIN = "wso2.com";
+
+    @DataProvider(name = "configProvider")
+    public static Object[][] configProvider() {
+        return new Object[][]{{TestUserMode.SUPER_TENANT_ADMIN}, {TestUserMode.TENANT_ADMIN}};
+    }
+
+    @Factory(dataProvider = "configProvider")
+    public OAuth2ServiceIntrospectionTestCase(TestUserMode userMode) throws Exception {
+
+        super.init(userMode);
+        AutomationContext context = new AutomationContext("IDENTITY", userMode);
+        this.username = context.getContextTenant().getTenantAdmin().getUserName();
+        this.userPassword = context.getContextTenant().getTenantAdmin().getPassword();
+        this.activeTenant = context.getContextTenant().getDomain();
+    }
 
     @BeforeClass(alwaysRun = true)
     public void testInit() throws Exception {
 
-        super.init(TestUserMode.SUPER_TENANT_USER);
         logManger = new AuthenticatorClient(backendURL);
         adminUsername = userInfo.getUserName();
         adminPassword = userInfo.getPassword();
-        logManger.login(isServer.getSuperTenant().getTenantAdmin().getUserName(),
-                isServer.getSuperTenant().getTenantAdmin().getPassword(),
-                isServer.getInstance().getHosts().get("default"));
+        logManger.login(username, userPassword, isServer.getInstance().getHosts().get("default"));
```

Comment:
change to 
 sessionCookie = logManger.login(username, userPassword, isServer.getInstance().getHosts().get("default"));